src/callbacks_stage/subcell_bounds_check_2d.jl

```diff
@@ -49,40 +39,35 @@
                 continue
             end
             key = Symbol(string(v), "_min")
-            deviation_threaded = idp_bounds_delta_local[key]
-            @threaded for element in eachelement(solver, cache)
-                deviation = deviation_threaded[stride_size * Threads.threadid()]
+            deviation = idp_bounds_delta_local[key]
+            @batch reduction=(max, deviation) for element in eachelement(solver, cache)
                 for j in eachnode(solver), i in eachnode(solver)
                     var = u[v, i, j, element]
                     deviation = max(deviation,
                                     variable_bounds[key][i, j, element] - var)
                 end
-                deviation_threaded[stride_size * Threads.threadid()] = deviation
             end
+            idp_bounds_delta_local[key] = deviation
         end
         for variable in limiter.positivity_variables_nonlinear
             key = Symbol(string(variable), "_min")
-            deviation_threaded = idp_bounds_delta_local[key]
-            @threaded for element in eachelement(solver, cache)
-                deviation = deviation_threaded[stride_size * Threads.threadid()]
+            deviation = idp_bounds_delta_local[key]
+            @batch reduction=(max, deviation) for element in eachelement(solver, cache)
```

Comment:
Same here

Reply:
Added a comment at the top in [51ad911](https://github.com/trixi-framework/Trixi.jl/pull/1888/commits/51ad911282dcc1f4704ee7b7a973fb98bcf7beba)